p2panda-rs/src/schema/field_types.rs

```diff
@@ -19,115 +22,134 @@ pub enum FieldType {
-            FieldType::PinnedRelationList => "pinned_relation_list",
+            FieldType::Bool => "bool".to_string(),
+            FieldType::Int => "int".to_string(),
+            FieldType::Float => "float".to_string(),
+            FieldType::String => "str".to_string(),
+            FieldType::Relation(schema_id) => format!("relation({})", schema_id.as_str()),
+            FieldType::RelationList(schema_id) => format!("relation_list({})", schema_id.as_str()),
+            FieldType::PinnedRelation(schema_id) => {
+                format!("pinned_relation({})", schema_id.as_str())
+            }
+            FieldType::PinnedRelationList(schema_id) => {
+                format!("pinned_relation_list({})", schema_id.as_str())
+            }
         }
     }
 }
 
+impl From<FieldType> for OperationValue {
+    fn from(field_type: FieldType) -> OperationValue {
+        OperationValue::Text(field_type.serialise())
+    }
+}
+
 impl FromStr for FieldType {
     type Err = FieldTypeError;
 
     fn from_str(s: &str) -> Result<Self, Self::Err> {
-        match s {
-            "bool" => Ok(FieldType::Bool),
-            "int" => Ok(FieldType::Int),
-            "float" => Ok(FieldType::Float),
-            "str" => Ok(FieldType::String),
-            "relation" => Ok(FieldType::Relation),
-            "relation_list" => Ok(FieldType::RelationList),
-            "pinned_relation" => Ok(FieldType::PinnedRelation),
-            "pinned_relation_list" => Ok(FieldType::PinnedRelationList),
-            type_str => Err(FieldTypeError::InvalidFieldType(type_str.into())),
+        // Matches a field type, followed an optional group in paranetheses that contains the
+        // referenced schema for relation field types.
+        let re = Regex::new(r"(\w+)(\((.+)\))?").unwrap();
+        let groups = re.captures(s).unwrap();
+
+        match (
+            groups.get(1).map(|m| m.as_str()),
+            groups.get(3).map(|m| m.as_str()),
+        ) {
+            (Some("bool"), None) => Ok(FieldType::Bool),
+            (Some("int"), None) => Ok(FieldType::Int),
+            (Some("float"), None) => Ok(FieldType::Float),
+            (Some("str"), None) => Ok(FieldType::String),
+            (Some("relation"), Some(schema_id)) => {
+                Ok(FieldType::Relation(SchemaId::new(schema_id).unwrap()))
+            }
+            (Some("relation_list"), Some(schema_id)) => {
+                Ok(FieldType::RelationList(SchemaId::new(schema_id).unwrap()))
+            }
+            (Some("pinned_relation"), Some(schema_id)) => {
+                Ok(FieldType::PinnedRelation(SchemaId::new(schema_id).unwrap()))
+            }
+            (Some("pinned_relation_list"), Some(schema_id)) => Ok(FieldType::PinnedRelationList(
+                SchemaId::new(schema_id).unwrap(),
+            )),
+            _ => Err(FieldTypeError::InvalidFieldType(s.into())),
         }
     }
 }
 
-impl From<FieldType> for String {
```

Comment:
I removed this conversion to more clearly separate serialisation and (debug) display of values.